### DISCUSSION TITLE
Uruchomienie sceny w domoticzu z przycisku (long press)

### DISCUSSION THREAD

**Discussion by kgraf80** [lkankowski/arduino-multi-relay2#44] [General]
Dzień dobry,
czy jest możliwość uruchomienia sceny (idx: 1) w domoticzu przez przyciśnięcie fizycznego przycisku (double-click lub long-press). Jeśli tak, to jak to zrobić?
Ps. dzięki za sketch
Pozdrawiam
Kamil

**Comment by lkankowski** [comment]
Hej,

Jeżeli dobrze Cię rozumiem, to masz fizyczny przycisk mono-stabilny, na którym chciałbyś mieć 2 akcje:
1. Po zwykłym kliknięciu zapewne ma się przełączyć przekaźnik (np. oświetlenie)
2. Dla long-press ma uruchomić się scena w Domoticzu.

Dla takiego scenariusza konfiguracja może wyglądać tak
```cpp
const RelayConfigDef gRelayConfig[] PROGMEM = {
  {1, 3, RELAY_TRIGGER_LOW, -1, "Przekaźnik 1"},
  {2, -7, RELAY_TRIGGER_LOW, -1, "Przekaźnik 2"},
};

const ButtonConfigDef gButtonConfig[] PROGMEM = {
  {A0, MONO_STABLE, 1, 2, -1, "Przycisk 1"},
);
```
W takiej konfiguracji "Przekaźnik 1" używa rzeczywistego GPIO 3 w Arduino, a "Przekaźnik 2" nie potrzebuje GPIO, więc wpisałem wartość ujemną (w tym wypadku "-7"), co oznacza "wirtualny pin". Jest to konieczne, gdyż sketch raportuje do MySensors, a więc też do Domoticza tylko przekaźniki (nie przyciski).

Po uruchomieniu na takiej konfiguracji, w Domoticzu zobaczysz "Przekaźnik 1" i "Przekaźnik 2". Teraz już możesz zrobić regułę, że po zmianie stanu "Przekaźnik 2" odpala się jakaś scena. Jak nie chcesz widzieć "Przekaźnik 2", to dodaj na początku nazwy "$".

**Reply by kgraf80** [reply]
Hej, dzięki za szybką i wyczerpującą odpowiedź. Wszystko działa jak powinno.
